contrib/python-bindings/source/cell_accessor_wrapper.cc

```diff
@@ -400,7 +400,7 @@ namespace python
 
   void CellAccessorWrapper::set_material_id(const int material_id)
   {
-    AssertThrow(material_id < numbers::invalid_material_id,
+    AssertThrow(static_cast<const types::material_id>(material_id) < numbers::invalid_material_id,
```

Comment:
Why do we store the material ID as an `int`?

Reply:
no idea, maybe @Rombur recalls

Reply:
I think we could store it as an `unsigned int` (or a `types::material_id`) but the range won't be the same as it is in C++. There is at least one mesh generator that uses a material ID of -1 for empty cells but boost throws when you try to extract a negative `unsigned int`, so I was forced to extract an `int`. That is why the material ID is an `int`.